### PR TITLE
MWT-24 Fixes

### PR DESCRIPTION
**Pull request by kaiwalyajoshi**
- Add support for mesos-agent-labels to spawn jenkins tasks on `windows` or `linux` agents with the default being `linux`.
- Update service options to reflect the changes done in version 4.0.0 of dcos-jenkins-service.
- Add group creation support to `sdk_marathon` to allow creation of quota-enforced `jenkins` group.
- Update security settings to use newer `mesos-jenkins` supported version of secrets.
- Change user of jenkins-master marathon app from `root` to `nobody`

### CODE REVIEW

File: conftest.py [hunk 1]
```diff
@@ -140,7 +140,13 @@ def pytest_addoption(parser):
         type=str,
         help="Select the containerizer used to launch the Jenkins master.",
     )
-
+    parser.addoption(
```
Comment (jeschkies) [inline conftest.py:143]: Is this really required? I don't mind but the service configures two Jenkins agent labels anyways.

**Review by jeschkies — CHANGES_REQUESTED** [review]
I don't think this addresses the issues of placing the Jenkins master, though. How can I force the master *not* to run on Windows? Please see https://jira.d2iq.com/browse/D2IQ-66928

**Review by jeschkies — APPROVED** [review]

**Comment by kaiwalyajoshi** [comment]
Everything looks :rocket: for MWT-24, just waiting on the results of the scale test before merging this PR!

File: conf/jenkins/configuration.yaml [hunk 2]
```diff
@@ -5,7 +5,7 @@ jenkins:
   numExecutors: 0
   clouds:
     - mesos:
-        agentUser: "${JENKINS_AGENT_USER:-root}"
+        agentUser: "${JENKINS_AGENT_USER:-nobody}"
```
Comment (jeschkies) [inline conf/jenkins/configuration.yaml:8]: @kaiwalyajoshi, I think the default should be `nobody`.

**Comment by kaiwalyajoshi** [comment]
Merging this branch into `4.x.y` as we were able to get 500 jenkins master and around 2k executors running.

**Pull request merged**